executor/union_scan.go

```diff
@@ -0,0 +1,243 @@
+package executor
```

Comment:
I will add tests for this later, but most of the code in this file is already covered by old tests.